lxd/instance.go

```diff
@@ -566,8 +570,15 @@ func pruneExpiredContainerSnapshotsTask(d *Daemon) (task.Func, task.Schedule) {
 					continue
 				}
 
+				snapshotID := snapshot.ID()
+				if cSnapshotsPruneRunning.Contains(strconv.Itoa(snapshotID)) {
+					continue
```

Comment:
I think the skipping should be handled in `pruneExpiredContainerSnapshots` not in `pruneExpiredContainerSnapshotsTask`. Basically all locking stuff should be in `pruneExpiredContainerSnapshots` where the actual call to `snapshot.Delete` is made.

Reply:
Everything related to locking  is in `pruneExpiredContainerSnapshots` right now.